restructuring documentation and updating information

choldgraf opened the pull request:
This is a medium-sized restructuring and update of the Binder user documentation. It has the following aims:

- Reorganize so we have less duplicated information and more natural clustering of information
- Put section files in folders so it's easier to track and maintain
- Update content about the deployments/federation, etc (closes #209 closes #195 )
- Update content in general to improve the guides and such (closes https://github.com/jupyterhub/binder/issues/200)
- Update the logo to make better use of logo space
- Add social media icons and links via the theme's config (closes #179)
- Uses sphinx panels to more nicely-organize many pages
- Updating more pages to markdown instead of rST

would love to hear thoughts! (cc @sgibson91 who helpfully opened several of the issues this tris to tackle...let me know if I missed something!)

consideRatio commented:
@choldgraf I think this looks great! I didn't find anything that I wanted to suggest an edit to. I didn't read through all the text, but found a lot of what I read very good. I think for example you did an excellent job in the about page giving a far more friendly introduction than the much more technical one that was around before.

For anyone else looking at this PR; don't miss out on readthedocs PR review feature that gives you a link in the commit checks!

betatim commented on doc/about/index.md:
```diff
@@ -0,0 +1,12 @@
+# mybinder.org info
```
```suggestion
# Information about mybinder.org
```

betatim commented on doc/about/user-guidelines.md:
```diff
@@ -0,0 +1,67 @@
+## Maximum Concurrent Users for a Service
+
+You may be considering using `mybinder.org` as a part of your service or product.
+Think that is great, and encourage people to experiment with using Binder technology
```
Missing words at the start of the sentence?

betatim commented on doc/about/user-guidelines.md:
```diff
@@ -0,0 +1,67 @@
+we've set a maximum limit of concurrent user sessions that point to the same
+Binder link. **The maximum number of simultaneous users for a given repo is 100**.
+If you think you have a *really* good reason for why this number should be
+higher, please [Open an Issue](https://github.com/jupyterhub/mybinder.org-deploy/issues).
```
Should we link to the same issue template as in the previous question? We'd want a lot of the same information as in the event case but the goal would be a permanent increase in quota.

Maybe we can get around having two templates or having to explain the difference between a temporary increase and a permanent one by rewording this question to be just about "what is the limit?" with an instruction to follow the link to the previous question if you want an increase. I think I like this idea because it has never happened that someone wanted a permanent increase (those all came through informal channels)

betatim commented:
This is a nice PR that implements a lot of issues which is great. It moves content and its adds new content. This makes it hard to see which is which. I'll try and leave a comment on things which I think are new content to call attention to it.

betatim commented on doc/_templates/edit_on_github.html:
```diff
@@ -1,12 +0,0 @@
-        </li>
-    </ul>
-     {% endif %}
-</div>
```
Why remove the "edit on github" buttons? I find them useful to find the repo and file that needs editing

choldgraf replied:
it's not removed - this is a template from an old theme that we were manually inserting, the pydata theme has an "edit this page" button built into it (see the previewed docs)

betatim commented on doc/about/about.rst:
```diff
@@ -1,72 +1,108 @@
-Frequently Asked Questions
```
If we move the page we need a redirect from the old URL to the new one.

choldgraf replied:
if we move all of these docs to binderhub does that mean we need to set up a redirect for every old doc page to every new doc page? Do we know how many direct links to binder docs there are? I suspect not that many...

betatim commented on doc/about/about.rst:
```diff
@@ -1,72 +1,108 @@
-Frequently Asked Questions
-==========================
```
What is the thinking behind merging the FAQ page and the about page?

choldgraf replied:
basically I was trying to separate out "content about preparing repositories from Binder" from "information specifically about `mybinder.org`". We already had a [FAQ-like about page](https://mybinder.readthedocs.io/en/latest/about.html) in our community section. In addition, most of the entries in the FAQ were actually about `mybinder.org` instead of building binder repositories, so I moved them over to the `mybinder.org-specific` section. Note that there *is* still a FAQ for the "Using" section, but it has been moved with the re-org: https://mybinder--213.org.readthedocs.build/en/213/using/faq.html

betatim commented on doc/about/user-guidelines.md:
```diff
@@ -0,0 +1,67 @@
+higher, please [Open an Issue](https://github.com/jupyterhub/mybinder.org-deploy/issues).
+to discuss with the community!
+
+## Maximum Concurrent Users for a Service
```
```suggestion
## Using mybinder.org as backend for your service

Please do not use mybinder.org as backend for your for-profit service or product.

The cloud resources consumed by mybinder.org are funded through donations or grants. Because
of this we can not subsidise for-profit products or services by offering free cloud resources to them.
This means we will prevent you from using mybinder.org as a service for your product.

Operating mybinder.org relies on the good will of volunteers and organisations to obtain the
resources required to do so. This is easier if we are not seen as supporting for-profit services
that are not a net contributor to the service.
```

What do you think of this? I thought that the new title was easier to find for someone thinking about providing a service based on mybinder.org. I think this also has a simpler message ("don't do it") instead of going into details of when/if/how you might run into trouble. I think the general principle is "don't do it" which is what readers should care about. This is because we might fix the load issues related to builds which might make you think it is now Ok because we said "builds cause trouble so we stop you". Whereas in reality it remains not Ok because the general rule is "don't do it'.

(this suggestion should probably replace all of this FAQ item)

choldgraf replied:
I think this is a reasonable explanation - I guess my main question is how we'd differentiate between "spacy-style backends" and "jovian.ml-style backends"? Is the decision question "do you directly make profit from running the service that Binder is powering for you?"

betatim commented:
It would be nice to get some feedback on https://github.com/jupyterhub/binderhub/issues/1091 as part of this docs rewrite. I think all documentation that isn't specific to mybinder.org should migrate to the BinderHub documentation.

sgibson91 commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
-.ipynb_checkpoints
\ No newline at end of file
+.ipynb_checkpoints
+.vscode
```
I recommend doing something [like this](http://egorsmirnov.me/2015/05/04/global-gitignore-file.html) so we don't end up maintaining an ignore list of everyone's favourite editors :)

sgibson91 commented on doc/about/about.rst:
```diff
@@ -1,72 +1,108 @@
+
+The technology behind mybinder.org is primarily composed of three open-source projects:
+
+* `JupyterHub <https://z2jh.jupyter.org>`_, which manages cloud infrastructure for user instances
```
Since we're stating that this is managing cloud infrastructure specifically, should we link to z2jh-k8s?

choldgraf replied:
Which link would you change? Isn't this already linking to that?

sgibson91 replied:
Yes sorry, my mistake!

sgibson91 commented on doc/about/about.rst:
```diff
@@ -1,72 +1,108 @@
+Is ``mybinder.org`` free to use?
+--------------------------------
+
+Yes! Though note that it has relatively :ref:`limited computational resources
```
```suggestion
Yes! Though note that it has :ref:`limited computational resources
```

I don't think this word is necessary. It begs the question "Relative to what?"

sgibson91 commented on doc/about/user-guidelines.md:
```diff
@@ -0,0 +1,67 @@
+
+:::{admonition,warning} When in doubt, open an issue
+If you're not sure whether your use-case is reasonable for `mybinder.org`, don't
+hesitate to [open an issue](https://github.com/jupyterhub/mybinder.org-deploy/issues/new?labels=impact&template=request_resources.md)
```
```suggestion
hesitate to [open an issue](https://github.com/jupyterhub/mybinder.org-deploy/issues/new?labels=question&template=ban_check.md)
```

Maybe the issue template to check if a repo meets the requirements is more appropriate here?

choldgraf replied:
I think this is a better template, but we should call it something other than `ban-check.md`. that will sound hostile to people that are just checking in on whether they have an OK use-case.

choldgraf replied:
see https://github.com/jupyterhub/mybinder.org-deploy/pull/1651 @sgibson91 WDYT?

sgibson91 replied:
Agreed. I'd clearly thought a bit about it since I gave it the title "repo check" but hadn't worked that idea all the way through 😂

sgibson91 reviewed (commented):
Thank you so much for this work @choldgraf! 🎉 ✨ I've left a few additional comments

choldgraf commented:
OK I believe that I've addressed all of the comments here - improved a bunch of the language per your suggestions and also added redirects for all of the moved pages with `sphinxext.rediraffe`.

@betatim I added some comments to the issue about moving the user docs to `/binderhub`...I think it could be a good idea! Perhaps something we can continue tackling after this round of improvements? I'm happy to keep iterating in the next few weeks.

consideRatio commented:
Great work on this everyone! 🥳😍

choldgraf commented:
Hey all, anything else for me to do on this one? (and independent of this, another note that @betatim is looking for feedback in https://github.com/jupyterhub/binderhub/issues/1091 on whether to combine the BinderHub user docs with the BinderHub admin docs)

sgibson91 reviewed (approved):
✨

choldgraf commented:
I see lots of 👍 and 🚀 and "approve"s, but nobody has pressed the green button!

consideRatio commented:
Thank you for your work om this @choldgraf and everyone for your review help! I love to see these kinds of improvements, but they are a bit scary to merge so i didnt do it myself before - i do think its important to help the PR authors move onwards though so i hit merge now - thanks for nudging with a comment!

choldgraf commented:
thanks @consideRatio!
